readthedocs/core/unresolver.py

```diff
@@ -82,6 +83,26 @@ def unresolve(self, url, add_index=True):
             path=parsed.path,
         )
 
+        # Make sure we are serving the external version from the subdomain.
+        if external_version_slug and version:
+            if external_version_slug != version.slug:
+                log.warning(
+                    "Invalid version for external domain.",
+                    domain=domain,
+                    version_slug=version.slug,
+                )
+                version = None
+                filename = None
```

Comment:
Setting version to `None` will be interpreted as like the version wasn't found, but we could return None, or raise an exception, since those URLs will be generated by manually changing the URL.

Reply:
I thought about this as well, and I'm not sure what's my position here.

I'd like to raise exceptions for all these cases were we return `None` for not found things. However, I'm not sure it's possible for all the cases because IIRC there are other places were `project, None, None` is useful to continue moving forward.

If that's not the case and I'm confused here, I'm 👍🏼 on standardizing this and always raise a detailed exception for each case that we can clearly handle from outside.

Reply:
Actually, I think the case that I'm referring to is the 404 pages (see https://github.com/readthedocs/readthedocs.org/pull/9534/files#diff-3f2b884fdd4752fe36587aee4530694e7a4ff10a8f4523a7b0bc25e9d2f11e11R162)

Reply:
yeah, basically this would be interpreted as a 404, like if the version wouldn't exist.